02-peer-protocol.md

```diff
@@ -607,13 +619,22 @@ A receiving node which closes an incoming HTLC in response to an
 `update-fail-htlc` message on an offered HTLC MUST copy the `reason`
 field to the outgoing `update-fail-htlc`.
 
+A receiving node MUST fail the channel if the `BADONION` bit in
+`failure-code` is not set for `update-fail-malformed-htlc`.
```

Comment:
If I remember correctly the `failure-code` is opaque to the nodes forwarding the error, due to encryption. So this would only affect the first hop, i.e., transfer origin to first node in the path, despite the failure creator setting these bits. It'd allow anybody on a path kill that first channel, potentially uncovering the true origin. The alternative is to push the `failure-code` into the cleartext, which has other issues.